extensions/spark/kyuubi-extension-spark-common/src/main/scala/org/apache/kyuubi/sql/zorder/ResolveZorderBase.scala

```diff
@@ -57,14 +61,43 @@ abstract class ResolveZorderBase extends Rule[LogicalPlan] {
   }
 
   override def apply(plan: LogicalPlan): LogicalPlan = plan match {
-    case statement: OptimizeZorderStatementBase if statement.query.resolved =>
+    case statement: OptimizeZorderStatement if statement.query.resolved =>
       checkQueryAllowed(statement.query)
       val tableIdentifier = getTableIdentifier(statement.tableIdentifier)
       val catalogTable = session.sessionState.catalog.getTableMetadata(tableIdentifier)
       buildOptimizeZorderCommand(catalogTable, statement.query)
 
     case _ => plan
   }
+
+  /**
+   * The `PredicateHelper.isLikelySelective()` is available since Spark-3.3, forked for Spark
+   * that is lower than 3.3.
+   * Forked from Apache Spark's org.apache.spark.sql.catalyst.expressions.PredicateHelper
+   *
+   * Returns whether an expression is likely to be selective
+   */
+  private def isLikelySelective(e: Expression): Boolean = e match {
+    case Not(expr) => isLikelySelective(expr)
+    case And(l, r) => isLikelySelective(l) || isLikelySelective(r)
```

Comment:
Seems not safe. We can check if the expression is `NamedExpression` and only allow the `Attribute`, and we can put the check inside `KyuubiSparkSQLParser`.

Reply:
After discussing with @ulysses-you  offline, we believe this verification should be safe.